client.go

```diff
@@ -176,10 +183,20 @@ func serveWs(server *Server, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client := &Client{server: server, conn: conn, send: make(chan []byte, 256), sendToken: make(chan SessionToken, 256)}
+	client := &Client{
+		server:    server,
+		conn:      conn,
+		send:      make(chan []byte, sendMsgBufferSize),
+		sendToken: make(chan SessionToken, sendTokenBufferSize),
+		start: make(chan struct{}),
+	}
+
 	client.sendToken <- SessionToken(sessionToken)
 	client.server.register <- client
 
-	go client.writePump()
-	go client.readPump()
+	func() {
+		<-client.start
```

Comment:
When this channel closes, the two go routines can become sending and receiving data from the client